addons/hr_timesheet/controllers/portal.py

```diff
@@ -123,11 +122,16 @@ def get_timesheets():
             timesheets = Timesheet_sudo.search(domain, order=orderby, limit=_items_per_page, offset=pager['offset'])
             if field:
                 if groupby == 'date':
```

Comment:
Why didn't you add the constraint in the domain then? Did I miss something? I am getting confused.

```suggestion
            if not timesheets:
                return timesheets, []
            domain = AND([domain, [('id', 'in', timesheets.ids)])
            if field:
                if groupby == 'date':
```

Reply:
I made this fix with NLE and to be honest he did most of the coding part, and I don't remember why he added this part to check the ids. (He's on leave for something like 2 weeks so I cannot ask him). But I think you'r right there is no need to check it.

Reply:
I think that it is indeed needed in case there would be a timesheet added in the database between the the search and the read_group. So adding the lines I provided you with with take that into account too, but without looping in the recordset of timesheets. Can you apply it?
Thanks 🙂